GridList/GridList.jsx

```diff
@@ -0,0 +1,178 @@
+          props[this._propsDict.secondaryActionIconAlign]
+      );
+    }
+
+    if (props[this._propsDict.aspectRatio]) {
+      classes.push(
+        "mdc-grid-list--tile-aspect-" + props[this._propsDict.aspectRatio]
+      );
+    }
+
+    if (props[this._propsDict.twoLineCaptions]) {
+      classes.push("mdc-grid-list--twoline-caption");
+    }
+
+    return classes.join(" ");
+  }
+  materialDom(props) {
+    const cols = props.cols || 2;
+    const tileWidthPct = Math.floor(100 / cols) + "%";
+
+    return (
+      <span>
+        <style
+          dangerouslySetInnerHTML={{
+            __html: `
+        .mdc-grid-list__tiles {
+          --mdc-grid-list-tile-width: ${tileWidthPct};
+        }
+        .mdc-grid-list__tiles > .mdc-grid-tile > * img {
+          width: 100%;
+        }
+      `
+          }}
+        />
+        <div
+          className={this.createClassName(props)}
+          ref={control => {
+            this.control = control;
+          }}
+          {...props}
+        >
+          <ul class="mdc-grid-list__tiles">{props.children}</ul>
+        </div>
+      </span>
+    );
+  }
+}
+
+/**
+ * @prop actionIcon	{Node} - An IconButton element to be used as secondary action target (primary action target is the tile itself).
+ * @prop actionPosition {'left'|'right'} - Position of secondary action IconButton.
+ * @prop children	{Node} - Theoretically you can pass any node as children, but the main use case is to pass an img, in whichcase GridTile takes care of making the image "cover" available space (similar to background-size: cover or to object-fit:cover).
+ * @prop cols	{number} - Width of the tile in number of grid cells.
+ * @prop containerElement	{string|node} - Either a string used as tag name for the tile root element, or a ReactElement. This is useful when you have, for example, a custom implementation of a navigation link (that knows about your routes) and you want to use it as the primary tile action. In case you pass a ReactElement, please ensure that it passes all props, accepts styles overrides and render it's children.
+ * @prop rows	{number} - Height of the tile in number of grid cells.
+ * @prop style {object} - Override the inline-styles of the root element.
+ * @prop subtitle	{Node} - String or element serving as subtitle (support text).
+ * @prop subtitleStyle {object} - Override the inline-styles of the subtitle element.
+ * @prop title {Node} - Title to be displayed on tile.
+ * @prop titleBackground {string} - 'rgba(0, 0, 0, 0.4)'	Style used for title bar background. Useful for setting custom gradients for example
+ * @prop titlePosition {'top'|'bottom'} - Position of the title bar (container of title, subtitle and action icon).
+ * @prop titleStyle {object} - Override the inline-styles of the title element.
+ */
+class GridListTile extends MaterialComponent {
+  constructor() {
+    super();
+    this.componentName = "grid-list-tile";
+    this._propsDict = {
+      actionIcon: "actionIcon",
```

Comment:
@filoozom shouldn't we implement this as a separate component?

Reply:
I think I'd prefer to have ```GridTile```s in a ```GridTileList``` that so that I could write something like:
```jsx
<GridTileList>
  <GridTile primary>
    <GridTile src="">
  </GridTile>
  <GridTile secondary>
    <GridTile title>Title</GridTile>
  </GridTile>
</GridTileList>
```

It would make the whole thing more modular. Haven't studied the doc yet though, so it might not make sense.

Reply:
This raises an important question for me: how much do we want to align with the Material UI version (http://www.material-ui.com/#/components/grid-list)?

An example from that page is:
```
   <GridList
      cellHeight={180}
      style={styles.gridList}
    >
      <Subheader>December</Subheader>
      {tilesData.map((tile) => (
        <GridTile
          key={tile.img}
          title={tile.title}
          subtitle={<span>by <b>{tile.author}</b></span>}
          actionIcon={<IconButton><StarBorder color="white" /></IconButton>}
        >
          <img src={tile.img} />
        </GridTile>
      ))}
    </GridList>
```

From this comment and the next, it sounds like the preference is for something like:
```
<GridList>
  {tilesData.map((tile) => (
      <GridTile key={tile.img} actionIcon={<IconButton><StarBorder color="white"/></IconButton>}>
             <GridTileTitle>{tile.title}</GridTileTitle>
             <GridTileSubTitle>
                <span>by <b>{tile.author}</b></span>
             </GridTileSubTitle>
             <img src={tile.img}/>
      </GridTile>
  ))}
</GridList>
```

Reply:
actually we're more focussed to be as close as possible to `mdc-web`. The project intends to be a thin declarative wrapper over it for preact.

Reply:
so am I to understand that the second example I gave is the tag structure you would prefer to use?

Reply:
Yes, we'd prefer the second one, with a ```GridTilePrimary``` for the ```img```.